gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +734,689 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
+/*                           GetNoDataValue()                           */
+/************************************************************************/
+
+double FITSRasterBand::GetNoDataValue( int * pbSuccess )
+
+{
+    poDS->LoadGeoreferencingAndPamIfNeeded();
+
+    if( bNoDataSet )
+    {
+        if( pbSuccess )
+            *pbSuccess = TRUE;
+
+        return dfNoDataValue;
+    }
+
+    if( poDS->bNoDataSet )
+    {
+        if( pbSuccess )
+            *pbSuccess = TRUE;
+
+        return poDS->dfNoDataValue;
+    }
+
+    return GDALPamRasterBand::GetNoDataValue( pbSuccess );
+}
+
+/************************************************************************/
+/*                           SetNoDataValue()                           */
+/************************************************************************/
+
+CPLErr FITSRasterBand::SetNoDataValue( double dfNoData )
+
+{
+    poDS->LoadGeoreferencingAndPamIfNeeded();
+
+    if( poDS->bNoDataSet && poDS->dfNoDataValue == dfNoData )
+    {
+        bNoDataSet = true;
+        dfNoDataValue = dfNoData;
+        return CE_None;
+    }
+
+    poDS->bNoDataSet = true;
+    poDS->dfNoDataValue = dfNoData;
+
+    poDS->bNoDataChanged = true;
+
+    bNoDataSet = true;
+    dfNoDataValue = dfNoData;
+    return CE_None;
+}
+
+/************************************************************************/
+/*                        DeleteNoDataValue()                           */
+/************************************************************************/
+
+CPLErr FITSRasterBand::DeleteNoDataValue()
+
+{
+    poDS->LoadGeoreferencingAndPamIfNeeded();
+
+    if( !poDS->bNoDataSet )
+        return CE_None;
+
+    poDS->bNoDataSet = false;
+    poDS->dfNoDataValue = -9999.0;
+
+    poDS->bNoDataChanged = true;
+
+    bNoDataSet = false;
+    dfNoDataValue = -9999.0;
+    return CE_None;
+}
+
+/************************************************************************/
+/*                     LoadGeoreferencingAndPamIfNeeded()               */
+/************************************************************************/
+
+void FITSDataset::LoadGeoreferencingAndPamIfNeeded()
+
+{
+    int status = 0;
+    int bitpix;
+    double dfScale, dfOffset;
+    double crpix1, crpix2, crval1, crval2, cdelt1, cdelt2, pc[4], cd[4];
+    double aRadius, cRadius, invFlattening = 0.0;
+    char target[16], ctype[9];
+    std::string pszGeogName, pszDatumName, projName;
+
+    const double PI = std::atan(1.0)*4;
+    const double DEG2RAD = PI / 180.;
+
+    if( !m_bReadGeoTransform && !m_bLoadPam )
+        return;
+
+/* -------------------------------------------------------------------- */
+/*      Get the transform from the FITS file.                           */
+/* -------------------------------------------------------------------- */
+    if( m_bReadGeoTransform )
+    {
+        m_bReadGeoTransform = false;
+
+        fits_read_key(hFITS, TSTRING, "OBJECT", target, nullptr, &status);
```

Comment:
which guarantee do we have that target won't be overflown ?

Reply:
I have modified `target` definition (https://github.com/epn-vespa/gdal/blob/c22caadc8e8df76430f53983d63bb9cb3fd0a517/gdal/frmts/fits/fitsdataset.cpp#L1332).
FITS header are 80 char long, the value of `OBJECT` can't have more than 80 char... then `status` will tell us if the file has something wrong.